bash_103_combining_commands/bash_103_combining_commands.md

```diff
@@ -0,0 +1,463 @@
+
+Maybe you want to know how many unique animals are in the file `last.txt`. We saw on the last page how to output the list of unique lines using `sort` followed by `uniq`. If we want to know how many (unique) lines there are, we can use the word count function `wc` with the option `-l` for number of lines:
+
+```
+sort last.txt | uniq | wc -l
+```
+
+This gives us the single numeric output `4`; there are 4 lines corresponding to the 4 unique animals listed in `last.txt`.
+
+What if we want to know which animal has the shortest scientific name? This also becomes a three step process:
+
+1. Count the characters in each `.txt` file with `wc -m *.txt`. For further explanation of how `*.txt` calls all of the `.txt.` files, see the [lesson on searching and organizing files](https://liascript.github.io/course/?https://raw.githubusercontent.com/arcus/education_modules/main/bash_command_line_102/bash_command_line_102.md).
+2. Sort the output by number with `sort -n`.
+3. Output the first line with `head -1`.
+
+Putting these steps together:
+
+```
+wc -m *.txt | sort -n | head -1
+      10 indian_cobra.txt
+```
+The Indian cobra, scientific name _Naja naja_ has the shortest character count of any of the animals listed here.
+
+<div class = "learnmore">
+**Why is the character count for `Naja naja` 10?**
+
+The character count includes each of the 8 letters, as well as the single blank space and the "newline" character that indicates the end of the line.
+</div>
+
+You can chain as many commands as you want into a pipeline, including search commands like `grep` and `find` that we haven't talked about in this lesson.
+
+### Quiz: linking with `|`
+
+Let's return to the uninformatively named `newfile.txt`:
+
+```
+Callinectes sapidus
+
+blue crab,crustacean
+
+Source: https://www.fisheries.noaa.gov/species/blue-crab
+
+The blue crab is a highly sought-after shellfish. Blue crabs live up and down the Atlantic Coast and in the Gulf of Mexico and are caught by both commercial and recreational fishermen. Its scientific name—Callinectes sapidus—translated from Latin means 'beautiful savory swimmer.'
+
+Blue crabs are the most valuable fishery in the Chesapeake Bay. They are also major predators of benthic communities and are prey for many other fish species. Blue crabs are so treasured in the region that the blue crab is the Maryland state crustacean.
+
+Length: 9 in
+Weight: .33 lbs
+```
+
+What command would append the 3rd line, which reads `blue crab,crustacean` to the file `Animals.csv`?
+
+[(X)] `head -3 newfile.txt | tail -1 >> Animals.csv`
+[( )] `head -3 newfile.txt | tail -1 > Animals.csv`
+[( )] `tail -1 newfile.txt | head -3 >> Animals.csv`
+[( )] `tail -1 newfile.txt | head -3 > Animals.csv`
+***
+<div class = "answer">
+The first three lines are returned by `head -3 newfile.txt`:
```

Comment:
Words are still getting squished together. I don't think newlines _within_ the highlight boxes are working well.

Reply:
seems they just get extra screwed up by not having the newlines at the start and end inside the tag.